packages/block-editor/src/components/provider/index.js

```diff
@@ -1,139 +1,88 @@
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
-import { withDispatch } from '@wordpress/data';
-import { compose } from '@wordpress/compose';
+import { useEffect, useRef } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import withRegistryProvider from './with-registry-provider';
 
-class BlockEditorProvider extends Component {
-	componentDidMount() {
-		this.props.updateSettings( this.props.settings );
-		this.props.resetBlocks( this.props.value );
-		this.attachChangeObserver( this.props.registry );
-	}
+function BlockEditorProvider( {
+	settings,
+	value,
+	children,
+	onChange = () => {},
+	onInput = () => {},
```

Comment:
Microoptimization: Instead of creating 2 separate functions these could both reference a dummy empty function.

Reply:
Specifically, the `noop` function provided by Lodash could be used here.